lib/spack/spack/build_systems/r.py

```diff
@@ -38,6 +44,61 @@ def configure_vars(self):
     def install(self, pkg, spec, prefix):
         """Installs an R package."""
 
+        try:
+            # TODO: use a more sustainable dcf parser
+            import pycran
+
+            # Read DESCRIPTION file with dependency information
+            # https://r-pkgs.org/description.html
+            r_deps = []
+            with open(fs.join_path(self.stage.source_path, 'DESCRIPTION')) as file:
+                for desc in pycran.parse(file.read()):
+                    if "Imports" in desc:
+                        r_deps.extend([d.strip() for d in desc["Imports"].split(",")])
+                    if "Depends" in desc:
+                        r_deps.extend([d.strip() for d in desc["Depends"].split(",")])
```

Comment:
Some packages (`leaps`) apparently have just `Depends: ` with nothing...
```suggestion
                    if "Imports" in desc:
                        r_deps.extend([d.strip() for d in desc["Imports"].split(",") if d != ""])
                    if "Depends" in desc:
                        r_deps.extend([d.strip() for d in desc["Depends"].split(",") if d != ""])
```